code/datums/components/signal_redirect.dm

```diff
@@ -0,0 +1,14 @@
+/datum/component/redirect
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	var/datum/callback/callback
+
+/datum/component/redirect/Initialize(list/signals, _callback)
+	//It's not our job to verify the right signals are registered here, just do it.
+	if(!signals || !_callback)
+		WARNING("A redirection component was initialized with incorrect args.")
+		return COMPONENT_INCOMPATIBLE
+	for(var/i in 1 to signals.len)
+		RegisterSignal(signals[i], .proc/relay_signal)
```

Comment:
This doesn't actually need RegisterSignal as all its doing is adding overhead to the call, what you should do is modify RegisterSignal to accept a proc path or a callback and then pass in the callback